# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: toolkit/library/rust/shared/Cargo.toml
# Commit: 5762b27d2cc8
# Full Hash: 5762b27d2cc82e6c7f3f0d37bc4bcfdb130f9220
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2023-06-21 04:00:08
# Description:
#   Bug 1839459 - Elide empty closed sub-paths instead of emitting a start point. r=jrmuizel
#   
#   There is a corner case that triggers an assert in wpf-gpu-raster if an empty sub-path
#   starts off a path. Just avoid the problem by eliding the sub-path, but at least handle
#   close correctly as per the Canvas2D closePath spec:
# ==============================================================================

diff -r caa96b0874f4 -r 5762b27d2cc8 toolkit/library/rust/shared/Cargo.toml
--- a/toolkit/library/rust/shared/Cargo.toml	Tue Jun 20 18:01:09 2023 +0000
+++ b/toolkit/library/rust/shared/Cargo.toml	Tue Jun 20 18:04:20 2023 +0000
@@ -94,7 +94,7 @@
 processtools = { path = "../../../components/processtools" }
 qcms = { path = "../../../../gfx/qcms", features = ["c_bindings", "neon"], default-features = false }
 
-wpf-gpu-raster = { git = "https://github.com/FirefoxGraphics/wpf-gpu-raster", rev = "112a2140c433af06938c663fc6777e2bf17bb6ce" }
+wpf-gpu-raster = { git = "https://github.com/FirefoxGraphics/wpf-gpu-raster", rev = "5ab6fe33d00021325ee920b3c10526dc8301cf46" }
 aa-stroke = { git = "https://github.com/FirefoxGraphics/aa-stroke", rev = "07d3c25322518f294300e96246e09b95e118555d" }
 
 # Force url to stay at 2.1.0. See bug 1734538.
